File: nodejs/src/utils/markdown.ts
import { marked } from 'marked';

// Escape raw HTML in markdown for security
marked.use({
  gfm: true,
  breaks: true,
  renderer: {
    html: ({ text }: { text: string }) =>
      text.replace(/&/g, '&amp;').replace(/</g, '&lt;').replace(/>/g, '&gt;')
  }
});

export function parseMarkdown(content: string): string {
  if (!content) return '';
  const result = marked.parse(content);
  return typeof result === 'string' ? result : '';
}

File: go/internal/pkg/utils/markdown.go
package utils

import (
	"github.com/gomarkdown/markdown"
)

func ParseMarkdown(content string) string {
	if content == "" {
		return ""
	}
	// Parse markdown to HTML
	result := markdown.ToHTML([]byte(content), nil, nil)
	html := string(result)

	// Use bluemonday to sanitize the HTML output
	return SanitizeHtml(html)
}
